packages/adapters/src/kubernetes-client-adapter.ts

```diff
@@ -0,0 +1,140 @@
+import { IObjectLogger } from "@scramjet/types";
+import * as k8s from "@kubernetes/client-node";
+import { ObjLogger } from "@scramjet/obj-logger";
+import { defer } from "@scramjet/utility";
+import { Writable, Readable } from "stream";
+import http from "http";
+
+class KubernetesClientAdapter {
+    logger: IObjectLogger;
+    name = "KubernetesClientAdapter";
+
+    private _configPath: string;
+    private _config?: k8s.KubeConfig
+    private _namespace: string;
+
+    constructor(configPath: string = "", namespace: string = "default") {
+        this.logger = new ObjLogger(this.name);
+        this._configPath = configPath;
+        this._namespace = namespace;
+    }
+
+    private get config(): k8s.KubeConfig {
+        if (!this._config) {
+            const kc = new k8s.KubeConfig();
+
+            try {
+                if (this._configPath && this._configPath.length) {
+                    kc.loadFromFile(this._configPath);
+                } else {
+                    kc.loadFromCluster();
+                }
+
+                this._config = kc;
+            } catch (err: any) {
+                this.logger.error("Unable to load kubeconfig", err);
+            }
+        }
+
+        if (!this._config) {
+            throw new Error("Kubernetes API client not initialized");
+        }
+
+        return this._config;
+    }
+
+    async createPod(metadata: k8s.V1ObjectMeta, spec: k8s.V1PodSpec, retries: number = 0) {
+        const kubeApi = this.config.makeApiClient(k8s.CoreV1Api);
+
+        const result = await this.runWithRetries(retries, "Create Pod", async () => {
+            return await kubeApi.createNamespacedPod(this._namespace, {
+                apiVersion: "v1",
+                kind: "Pod",
+                metadata,
+                spec
+            });
+        });
```

Comment:
Looks like a nice utility function. Maybe some refactoring should be done.

Also `return await` doesn't allow promises to chain in tail call and I don't really see a benefit here?

```suggestion
        const result = await this.runWithRetries(retries, "Create Pod", async () => 
            kubeApi.createNamespacedPod(this._namespace, {
                apiVersion: "v1",
                kind: "Pod",
                metadata,
                spec
            }));
```

Reply:
fixed